core/domain/blog_services.py

```diff
@@ -794,3 +836,96 @@ def update_blog_models_author_and_published_on_date(
+            if blog_post_summary is not None
+        ])
+
+
+def get_blog_post_ids_matching_query(
+    query_string: str, tags: List[str], offset: Optional[int]=None
+) -> Tuple[list[str], Optional[int]]:
+    """Returns a list with all blog post ids matching the given search query
+    string, as well as a search offset for future fetches.
+
+    This method returns exactly
+    feconf.MAX_NUM_CARDS_TO_DISPLAY_ON_BLOG_SEARCH_RESULTS_PAGE results if
+    there are at least that many, otherwise it returns all remaining results.
+    (If this behaviour does not occur, an error will be logged.) The method
+    also returns a search offset.
+
+    Args:
+        query_string: str. A search query string.
+        tags: list(str). The list of tags to query for. If it is empty, no tags
+            filter is applied to the results. If it is not empty, then a result
+            is considered valid if it matches at least one of these tags.
+        offset: int or None. Optional offset from which to start the search
+            query. If no offset is supplied, the first N results matching
+            the query are returned.
+
+    Returns:
+        2-tuple of (returned_blog_post_ids, search_offset). Where:
+            returned_blog_post_ids : list(str). A list with all
+                blog post ids matching the given search query string,
+                as well as a search offset for future fetches.
+                The list contains exactly
+                feconf.MAX_NUM_CARDS_TO_DISPLAY_ON_BLOG_SEARCH_RESULTS_PAGE
+                results if there are at least that many, otherwise it
+                contains all remaining results. (If this behaviour does
+                not occur, an error will be logged.)
+            search_offset: int. Search offset for future fetches.
+    """
+    returned_blog_post_ids: list[str] = []
+    search_offset: Optional[int] = offset
+
+    for _ in range(MAX_ITERATIONS):
+        remaining_to_fetch = (
+            feconf.MAX_NUM_CARDS_TO_DISPLAY_ON_BLOG_SEARCH_RESULTS_PAGE -
+            len(returned_blog_post_ids)
+        )
+
+        blog_post_ids, search_offset = (
+            search_services.search_blog_post_summaries(
+            query_string, tags, remaining_to_fetch, offset=search_offset))
+
+        invalid_blog_post_ids = []
+        for ind, model in enumerate(
+                blog_models.BlogPostSummaryModel.get_multi(blog_post_ids)):
+            if model is not None:
+                returned_blog_post_ids.append(blog_post_ids[ind])
+            else:
+                invalid_blog_post_ids.append(blog_post_ids[ind])
+
+        if (
+            (
+                len(returned_blog_post_ids) ==
+                feconf.MAX_NUM_CARDS_TO_DISPLAY_ON_BLOG_SEARCH_RESULTS_PAGE
+            ) or search_offset is None
+        ):
+            break
+
+        logging.error(
+                'Search index contains stale blog post ids: %s' %
+                ', '.join(invalid_blog_post_ids))
```

Comment:
```suggestion
        logging.error(
            'Search index contains stale blog post ids: %s' %
            ', '.join(invalid_blog_post_ids))
```

Should this only be printed when `len(invalid_blog_post_ids) > 0`

Reply:
Yup, Added if statement! Thanks!